hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/HoodieMetrics.java

```diff
@@ -207,6 +210,13 @@ public Timer.Context getIndexCtx() {
     return indexTimer == null ? null : indexTimer.time();
   }
 
+  public Timer.Context getPreWriteTimerCtx() {
+    if (config.isMetricsOn() && preWriteTimer == null) {
+      preWriteTimer = createTimer(preWriteTimerName);
+    }
```

Comment:
`preWrite` is not a meaningful terminology, don't think we should expose an metrics for it.

Reply:
Looks like a indexing time for the records, how about name it as just "indexing".

Reply:
@danny0405 this metric also contains the latency of reading the data from source. So should we rename it to `reading_source_and_indexing`?

Reply:
sure. sg

Reply:
+1 for `source_read_and_index`.